components/http/okHttp/src/main/java/com/microsoft/kiota/http/OkHttpRequestAdapter.java

```diff
@@ -306,243 +291,222 @@ private String getMediaTypeAndSubType(@Nonnull final MediaType mediaType) {
-                        try(final Scope deserializationScope = deserializationSpan.makeCurrent()) {
-                            final Object result = rootNode.getCollectionOfEnumValues(targetClass);
-                            setResponseType(result, span);
-                            return CompletableFuture.completedFuture((List<ModelType>)result);
-                        } finally {
-                            deserializationSpan.end();
-                        }
-                    } catch(ApiException ex) {
-                        return new CompletableFuture<List<ModelType>>(){{
-                            this.completeExceptionally(ex);
-                        }};
-                    } catch(IOException ex) {
-                        return new CompletableFuture<List<ModelType>>(){{
-                            this.completeExceptionally(new RuntimeException("failed to read the response body", ex));
-                        }};
+            Response response = this.getHttpResponseMessage(requestInfo, span, span, null);
+            final ResponseHandler responseHandler = getResponseHandler(requestInfo);
+            if(responseHandler == null) {
+                boolean closeResponse = true;
+                try {
+                    this.throwIfFailedResponse(response, span, errorMappings);
+                    if(this.shouldReturnNull(response)) {
+                        return null;
+                    }
+                    final ParseNode rootNode = getRootParseNode(response, span, span);
+                    if (rootNode == null) {
+                        closeResponse = false;
+                        return null;
+                    }
+                    final Span deserializationSpan = GlobalOpenTelemetry.getTracer(obsOptions.getTracerInstrumentationName()).spanBuilder("getCollectionOfEnumValues").setParent(Context.current().with(span)).startSpan();
+                    try(final Scope deserializationScope = deserializationSpan.makeCurrent()) {
+                        final Object result = rootNode.getCollectionOfEnumValues(targetClass);
+                        setResponseType(result, span);
+                        return (List<ModelType>)result;
                     } finally {
-                        closeResponse(closeResponse, response);
+                        deserializationSpan.end();
                     }
-                } else {
-                    span.addEvent(eventResponseHandlerInvokedKey);
-                    return responseHandler.handleResponseAsync(response, errorMappings);
+                } finally {
+                    closeResponse(closeResponse, response);
                 }
-            });
+            } else {
+                span.addEvent(eventResponseHandlerInvokedKey);
+                return responseHandler.handleResponse(response, errorMappings);
+            }
+
+
+        } catch(ApiException ex) {
```

Comment:
@andreaTP I'd like your opinion here. All generated exceptions derive from APIexception, which is checked.
Do you think we should:
- wrap the exception in a runtime one like we're doing here? (least favorite)
- declare the exception on the API surface (for the request adapter interface, the implementation, and the request executors), which would lead to consumers having to handle that exception?
- change the exception to be unchecked?

Reply:
Thanks a lot for sharing @baywet !

On this subject, I do have a strong opinion, but it might or might not be aligned with the rest of the Java community.
I would:

> change the exception to be unchecked

Among the top motivations:

- Checked Exceptions do not play well with lambdas, functional interface, and most of the more "modern" std lib API constructs
- Checked Exceptions will cause major disruption when an OpenAPI changes (e.g. if you declare an additional error mapping you have to propagate it in all of the generated and the code using it)

I can provide more evidence if needed/required, but let's try to get @EricWittmann feedback here as well.

Reply:
to your second point, since all mapped errors derive from Api Exception, wouldn't declaring ApiException be enough in the case we stick with checked exception?

Reply:
correct, that's possible, but it depends on the user code.

Reply:
For my money it's unchecked exceptions all the way for code like this.  So much easier to work with, and you can still catch whatever exceptions you want.  It might make it **slightly** less explicit but so much easier to work with.  Just my $0.02.  :)

Reply:
alright, thanks for the input here, I think we'll rely on your domain knowledge on this one.
@ramsessanchez can you:
- update ApiException to derive from an unchecked exception instead.
- update the code to stop wrapping up ApiException (or derived from generated code) in runtime exception.
- remove any throws ApiException from the method declarations.
Please?